components/hierarchical-view/hierarchical-view-mixin.js

```diff
@@ -472,4 +505,28 @@ export const HierarchicalViewMixin = superclass => class extends superclass {
 		this.__resizeObserver.disconnect();
 		this.__resizeObserver.observe(content);
 	}
+
+	__updateFocusables(view, sourceAttribute, targetAttribute) {
+		const rootView = this.getRootView();
+		let node = getComposedParent(view);
+		while (node && node !== rootView) {
+			if (node.nodeType === Node.ELEMENT_NODE) {
+				const tabIndex = node.getAttribute(sourceAttribute);
+				if (tabIndex !== null) {
+					node.setAttribute(targetAttribute, tabIndex);
+					node.removeAttribute(sourceAttribute);
+				}
+			}
+			node = getComposedParent(node);
+		}
+	}
+
+	__updateFocusablesHide(view) {
+		this.__updateFocusables(view, 'data-d2l-hierarchical-view-tabindex', 'tabindex');
+	}
+
+	__updateFocusablesShow(view) {
+		this.__updateFocusables(view, 'tabindex', 'data-d2l-hierarchical-view-tabindex');
+	}
```

Comment:
The names seem backwards to me here, but maybe I'm not understanding.  `__updateFocusablesHide` adds/shows the focus back, and `__updateFocusablesShow` removes/hides the focus again?

Reply:
Or maybe the names are strictly tied to updating the focus because of what is happening? I.e. when we are now showing a new view, we run `__updateFocusablesShow` (and hide the old ones), and when we hide the view we run `__updateFocusablesHide` (which shows the new ones).

Reply:
>Or maybe the names are strictly tied to updating the focus because of what is happening?

Yeah, `__updateFocusablesHide` is for when a view is being hidden (drilling up), whereas `__updateFocusablesShow` is for when a view is being shown (drilling in).  I think different names would be better... to reflect what it is doing as opposed to what is happening.  Maybe, `enableFocusableAncestors`/`disableFocusableAncestors`.  I originally considered those, but thought enable/disabled wasn't quite right.  Still better in hindsight though.  Thoughts?

Reply:
Maybe `reenable/disable` or `readd/remove`?  Or a better word that doesn't include "re", but means "again"?

Reply:
Ok, I updated to `__removeAncestorTabIndicies` and `__resetAncestorTabIndicies`... I _think_ that is clearer/less confusing.

Reply:
Yup I like it! 👍